addons/beehave/debug/tree_node.gd

```diff
@@ -140,8 +138,9 @@ func _check_all_children_on_screen(node: TreeNode) -> void:
 
 
 func _check_for_conflicts(node: TreeNode) -> void:
-	var min_distance = tree_distance + node.item.size.x
+	var min_distance = sibling_distance
 	var shift_value: float = 0
+	var shift_sibling: TreeNode = null
```

Comment:
Calculate the last sibling that pushes us to the right. Then we are sure that all other siblings between us and `shift_sibling` can be centered without needing to check for conflicts again